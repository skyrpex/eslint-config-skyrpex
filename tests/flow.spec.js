const { createEngine, cleanReport } = require("./utils");

const engine = createEngine({
    configFile: require.resolve("../flow.js"),
});

test("should report incompatible type error", () => {
    const report = engine.executeOnFiles(["incompatible-type.fixture.js"]);
    expect(cleanReport(report)).toMatchSnapshot();
});
