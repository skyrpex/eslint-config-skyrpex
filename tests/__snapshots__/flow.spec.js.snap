// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should report incompatible type error 1`] = `
Object {
  "errorCount": 1,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "results": Array [
    Object {
      "errorCount": 1,
      "filePath": "tests/fixtures/incompatible-type.fixture.js",
      "fixableErrorCount": 0,
      "fixableWarningCount": 0,
      "messages": Array [
        Object {
          "column": 21,
          "endColumn": 26,
          "endLine": 6,
          "line": 6,
          "message": "Cannot assign \`\\"123\\"\` to \`this.prop\` because string is incompatible with number (see line 3).",
          "nodeType": null,
          "ruleId": "flowtype-errors/show-errors",
          "severity": 2,
          "source": "        this.prop = \\"123\\";",
        },
      ],
      "source": "// @flow
class MyClass {
    prop: number;

    method() {
        this.prop = \\"123\\";
    }
}

const myClass = new MyClass();
myClass.method();
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;
